File: pdu/python/sensor_msgs/pdu_cdr_conv_ChannelFloat32.py
from .pdu_pytype_ChannelFloat32 import ChannelFloat32
from ..pdu_cdr_runtime import CdrReader, CdrWriter




def py_to_cdr_body_ChannelFloat32(writer: CdrWriter, src: ChannelFloat32):
    writer.write_string(src.name)
    writer.write_sequence_length(src.values)
    for elem in src.values:
        writer.write_float32(elem)


def cdr_body_to_py_ChannelFloat32(reader: CdrReader, dst: ChannelFloat32):
    dst.name = reader.read_string()
    dst.values = [reader.read_float32() for _ in range(reader.read_uint32())]
    return dst


def py_to_cdr_ChannelFloat32(src: ChannelFloat32) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_ChannelFloat32(writer, src)
    return writer.bytes()


def cdr_to_py_ChannelFloat32(cdr_payload) -> ChannelFloat32:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = ChannelFloat32()
    return cdr_body_to_py_ChannelFloat32(reader, dst)

File: pdu/javascript/sensor_msgs/pdu_cdr_conv_ChannelFloat32.js
import { ChannelFloat32 } from './pdu_jstype_ChannelFloat32.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';


export class PduChannelFloat32Converter {
    /**
     * @param {PduCdrWriter} writer
     * @param { ChannelFloat32 } src
     */
    static to_cdr_body(writer, src) {
        writer.write_string(src.name);
        writer.write_sequence_length(src.values);
        for (const elem of src.values) {
            writer.write_float32(elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { ChannelFloat32 } dst
     * @returns { ChannelFloat32 }
     */
    static cdr_body_to_js(reader, dst) {
        dst.name = reader.read_string();
        dst.values = Array.from({ length: reader.read_uint32() }, () => reader.read_float32());
        return dst;
    }

    /**
     * @param { ChannelFloat32 } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { ChannelFloat32 }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new ChannelFloat32());
    }
}
